### ISSUE TITLE
check for dataset locks?

### ISSUE THREAD

**Issue by donsizemore** [gdcc/python-dvuploader#9]
during bulk upload, a dataset may be locked for file ingest. Juan Pablo at IQSS is checking for locks and waiting them out: https://github.com/IQSS/dataverse-sample-data/blob/bd2f6bda7cab0475183466a13b09889fe606d3e0/create_sample_data.py#L19

would this be appropriate for python-dvuploader?
(if you already do this and I'm just making noise, feel free to close the issue without a response)

**Comment by JR-1991** [comment]
@donsizemore thanks for the suggestion! Have not added this yet and it is a great idea. Will add it asap to the library 🙌

**Comment by JR-1991** [comment]
Indeed, it would be beneficial, but only in the case of non-direct uploads. Direct uploads are direct communications with S3 and are not susceptible to lock issues caused by ingestions. The final „register“ step is performed through a batch interface, resulting in a single call for all uploaded files instead of individual calls.

However, there have been reports that within the Dataverse instance, optimistic lock exceptions can cause problems. This is documented [here](https://github.com/gdcc/python-dvuploader?tab=readme-ov-file#troubleshooting).

For non-direct uploads, I’ve added retry logic that addresses general issues (e.g., locks, network issues) and can be configured according to your needs. You can find more information about this in [here](https://github.com/gdcc/python-dvuploader?tab=readme-ov-file#environment-variables). This might help you tweak it further. If not, I’m happy to add additional safeguards using lock checks.

Do you experience this issue for both direct and non-direct uploads? I’d be happy to help.
